ui/kotlinx-coroutines-android/src/HandlerDispatcher.kt

```diff
@@ -33,15 +33,12 @@ public sealed class HandlerDispatcher : MainCoroutineDispatcher(), Delay {
 
 @Keep
 internal class AndroidDispatcherFactory : MainDispatcherFactory {
-    companion object {
-        @JvmStatic // accessed reflectively from core
-        fun getDispatcher(): MainCoroutineDispatcher = Main
-    }
 
-    override fun createDispatcher(): MainCoroutineDispatcher = Main
+    override fun createDispatcher(allFactories: List<MainDispatcherFactory>) =
+        HandlerContext(Looper.getMainLooper().asHandler(async = true), "Main")
 
     override val loadPriority: Int
-        get() = Int.MAX_VALUE
+        get() = Int.MAX_VALUE / 2
```

Comment:
Why do you think it will become part of usable API?
`MainDispatcherFactory` is an internal mechanism and should not be used externally.

Reply:
That's fair. I imagine someone will end up using it since it is possible - though I can't think of a use case. Reasonable to leave intentionally undocumented.